File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        if(len(cost)==1):
            return cost[0]
        if(len(cost)==2):
            return min(cost[0],cost[1])
        prev1=0
        prev2=0
        
        for i in range(2,len(cost)+1):
            one=prev1+cost[i-1]
            two=prev2+cost[i-2]

            res=min(one,two)
            prev2=prev1
            prev1=res
        return prev1


        

File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int[] dp=new int[cost.length+1];

        dp[0]=0;
        dp[1]=0;
        for(int i=2;i<=cost.length;i++)
        {
            int f=dp[i-1]+cost[i-1];
            int l=dp[i-2]+cost[i-2];

            dp[i]=Math.min(f,l);
        }
        return dp[cost.length];
    }
}